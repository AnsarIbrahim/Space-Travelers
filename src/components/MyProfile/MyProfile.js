import React from 'react';
import { useSelector } from 'react-redux';

export default function MyProfile() {
  const { rockets } = useSelector((store) => store.rocket);
  const filteredRockets = rockets.filter((rocket) => rocket.reserved === true);
  const { missions } = useSelector((store) => store.mission);
  const filteredMission = missions.filter((mission) => mission.reserved === true);
  return (
    <div className="pro-container">
      <div className="dragons-list">
        <h4 className="profile-header">My Rockets</h4>
        <table className="profile_table">
          <tbody className="body">
            { filteredRockets.length === 0 && <p>You have no reserved rockets</p>}
            {filteredRockets.map((rocket) => {
              const rocketId = rocket.id;
              return (
                <tr key={rocketId}>
                  <td className="text">{rocket.name}</td>
                </tr>
              );
            })}
          </tbody>
        </table>
      </div>
      <div className="dragons-list">
        <h4 className="profile-header">My Missions</h4>
        <table className="profile_table">
          <tbody className="body">
            { filteredMission.length === 0 && <p>You have no reserved rockets</p>}
            {filteredMission.map((mission) => {
              const missionId = mission.id;
              return (
                <tr key={missionId}>
                  <td className="text">{mission.name}</td>
                </tr>
              );
            })}
          </tbody>
        </table>
      </div>
    </div>
  );
}
