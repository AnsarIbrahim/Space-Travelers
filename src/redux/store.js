import { configureStore } from '@reduxjs/toolkit';
import missionReducer from './missions/missionsSlice';
import rocketReducer from './rockets/rocketsSlice';

const store = configureStore({
  reducer: {
    mission: missionReducer,
    rocket: rocketReducer,
  },
});

export default store;
