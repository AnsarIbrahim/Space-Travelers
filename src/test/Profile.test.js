import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import MyProfile from '../components/MyProfile/MyProfile';
import store from '../redux/store';

it('renders correctly', () => {
  const profile = renderer
    .create(
      <Provider store={store}><MyProfile /></Provider>,
    )
    .toJSON();
  expect(profile).toMatchSnapshot();
});
