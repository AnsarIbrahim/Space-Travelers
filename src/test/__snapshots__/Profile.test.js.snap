// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="pro-container"
>
  <div
    className="dragons-list"
  >
    <h4
      className="profile-header"
    >
      My Rockets
    </h4>
    <table
      className="profile_table"
    >
      <tbody
        className="body"
      >
        <p>
          You have no reserved rockets
        </p>
      </tbody>
    </table>
  </div>
  <div
    className="dragons-list"
  >
    <h4
      className="profile-header"
    >
      My Missions
    </h4>
    <table
      className="profile_table"
    >
      <tbody
        className="body"
      >
        <p>
          You have no reserved rockets
        </p>
      </tbody>
    </table>
  </div>
</div>
`;
